docs/extensions/extension.md

```diff
@@ -0,0 +1,92 @@
+# Contract: `Extension` resource
+
+Gardener defines common procedures which must be passed to get a functioning shoot cluster. Well known steps are represented by special resources like `Infrastructure`, `OperatingSystemConfig` or `DNS`. These resources are typically reconciled by dedicated controllers setting up the infrastructure on the hyperscaler or managing DNS entries, etc..
+
+But, some requirements don't match with those special resources or don't depend on being proceeded at a specific step in the creation / deletion flow of the shoot. They require a more generic hook. Therefore, Gardener offers the `Extension` resource.
+
+## What is required to register and support an Extension type
+
+Gardener creates one `Extension` resource per registered extension type in `ControllerRegistration` per shoot. 
+
+```yaml
+apiVersion: core.gardener.cloud/v1alpha1
+kind: ControllerRegistration
+metadata:
+  name: extension-example
+spec:
+  resources:
+  - kind: Extension
+    type: example
+    globallyEnabled: true
+```
+
+If `spec.resources[].globallyEnabled` is `true` the `Extension` resources of the given `type` is created for every shoot cluster. Set to `false`, the `Extension` resource is only created if configured in the `Shoot` manifest.
+
+The `Extension` resources is created in the shoot namespace of the seed cluster.
+
+```yaml
+---
+apiVersion: extensions.gardener.cloud/v1alpha1
+kind: Extension
+metadata:
+  name: example
+  namespace: shoot--foo--bar
+spec:
+  type: example
+  providerConfig: {}
+```
+
+Your controller needs to reconcile `extensions.extensions.gardener.cloud`. Since there can exist multiple `Extension` resources per shoot, each one holds a `spec.type` field to let controllers check their responsibility.
```

Comment:
```suggestion
Your controller needs to reconcile `extensions.extensions.gardener.cloud`. Since there can exist multiple `Extension` resources per shoot, each one holds a `spec.type` field to let controllers check their responsibility (similar to all other extension resources of Gardener).
```